script/Distribute.s.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+// Copyright (C) 2024 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.21;
+
+import "forge-std/Script.sol";
+
+import { ScriptTools } from "dss-test/ScriptTools.sol";
+import { Domain } from "dss-test/domains/Domain.sol";
+
+interface DistributionLike {
+    function distribute() external returns (uint256);
+}
+
+interface L1ProxyLike {
+    function estimateDepositCost(uint256, uint256, uint256) external view returns (uint256, uint256);
+}
+
+// Run vestedRewardsDistribution.distribute() to test deployement
+contract Distribute is Script {
+    using stdJson for string;
+
+    uint256 l1PrivKey = vm.envUint("L1_PRIVATE_KEY");
+
+    function run() external {
+        StdChains.Chain memory l1Chain = getChain(string(vm.envOr("L1", string("mainnet"))));
+        vm.setEnv("FOUNDRY_ROOT_CHAINID", vm.toString(l1Chain.chainId)); // used by ScriptTools to determine config path
+        string memory config = ScriptTools.loadConfig("config");
+        string memory deps   = ScriptTools.loadDependencies();
+        Domain l1Domain = new Domain(config, l1Chain);
+        l1Domain.selectFork();
+       
+        DistributionLike distribution = DistributionLike(deps.readAddress(".vestedRewardsDistribution"));
+
+        vm.startBroadcast(l1PrivKey);
+        distribution.distribute();
```

Comment:
IDK, it worked ok during testing, and since this is sending only one tx I think the gas params can be modified from the command line if needed.

Reply:
Can the gas limit really be changed via the CLI for a script run? Didn't seem possible when I checked. 

It will work until there is an L2 gas spike and then it will fail. Probably fine for testnet as you can just retry it but could waste gas in case this script is ran on L1 for whatever reason. I guess I'd at least add a comment in the script to point out that the call could ran out of gas.

Reply:
Found the `--gas-estimate-multiplier` option and added 20% in the instructions.

Reply:
Oh nice find, resolving